live-build/misc/upgrade-scripts/verify

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+. "${BASH_SOURCE%/*}/common.sh"
+
+function usage() {
+	echo "$(basename "$0"): $*" >&2
+	echo "Usage: $(basename "$0") -v <version>"
+	exit 2
+}
+
+function cleanup() {
+	#
+	# Capture the exit code here, and use that to determine if
+	# upgrade verification was successful or not.
+	#
+	local rc="$?"
+
+	#
+	# If the CONTAINER variable is empty, this means container
+	# creation failed. In that case, the container creation process
+	# will have already cleaned up after itself, and there's no
+	# further cleanup for us to do here.
+	#
+	[[ -z "$CONTAINER" ]] && return
+
+	#
+	# On failure, and when DLPX_DEBUG is true, we avoid cleaning up
+	# the verification container. This is intended to make it easier
+	# to diagnose and debug verification failures on developer
+	# machines.
+	#
+	if [[ $rc -ne 0 ]] && [[ -n "$DLPX_DEBUG" ]] && $DLPX_DEBUG; then
+		report_progress_inc 80 "Skipping upgrade verification cleanup"
+	else
+		report_progress_inc 80 "Performing upgrade verification cleanup"
+
+		"${BASH_SOURCE%/*}/verify-container" stop "$CONTAINER"
```

Comment:
nit: here and elsewhere, perhaps replace `${BASH_SOURCE%/*}` by a variable defined in `common.sh`?

Reply:
We'd still have to use `${BASH_SOURCE%/*}` when sourcing `common.sh`.. as a result, I think it's fine to just use it throughout, but I don't feel strongly.. I can make that change if you think it'd make things more readable.

It's worth noting though, `${BASH_SOURCE%/*}` isn't necessarily the same when used in `common.sh`. Practically it is, since `common.sh` and the scripts are in the same directory, but if we later were to move `common.sh` that'd break.

Reply:
You are right, it's probably better to have something like:
```
TOP="${BASH_SOURCE%/*}"
. "${TOP}/common.sh"
```